src/Avalonia.Controls/ToolTipService.cs

```diff
@@ -102,9 +103,31 @@ private void ControlPointerEntered(object? sender, PointerEventArgs e)
         /// </summary>
         /// <param name="sender">The event sender.</param>
         /// <param name="e">The event args.</param>
-        private void ControlPointerExited(object? sender, PointerEventArgs e)
+        private async void ControlPointerExited(object? sender, PointerEventArgs e)
         {
             var control = (Control)sender!;
+            var toolTip = control.GetValue(ToolTip.ToolTipProperty);
+            
+            // wait for the tooltip to activate
+            await Task.Delay(100).ConfigureAwait(true);
+            
+            if (control.IsPointerOver) return;
+            if (toolTip?.IsPointerOver == true)
```

Comment:
The idea is to check if the mouse is over the original control or over the tooltip. The problem is that there is a slight delay between these two actions.